src/jrd/tra.cpp

```diff
@@ -143,7 +143,13 @@ void TRA_attach_request(Jrd::jrd_tra* transaction, Jrd::jrd_req* request)
 void TRA_detach_request(Jrd::jrd_req* request)
 {
 	if (!request->req_transaction)
+	{
+		fb_assert(!request->req_savepoints);
 		return;
+	}
+
+	// Release stored looper savepoints
+	Savepoint::destroy(request->req_savepoints);
```

Comment:
On 06/14/2016 04:47 PM, Dimitry Sibiryakov wrote:
>>   		return;
>> +	}
>> +
>> +	// Release stored looper savepoints
>> +	Savepoint::destroy(request->req_savepoints);
> For this block 'pool' pointer is invalid, 'next' pointer is valid, but code handle it as a pointer to pool, that's a reason for crash.

That means that code tries to free some block once again.

> Perhaps, single savepoints are not supposed to be freed at all, they will die at once with transaction pool...
>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/FirebirdSQL/firebird/pull/29/files/057fc3a9ee0a851a39de07ad6497b28ac48ce931#r66972581